Added delete comment function

MattLBC opened the pull request:
Added delete function to comments 
Added single error page for any incorrect path

cjplatten commented:
Functionality works well and good UI response! 
The only thing I'd say to consider right now is that currently any user can delete any comment, while it's not compulsory to user a userContext that could be something to consider any would allow you to block users being able to delete other users' comments.

MattLBC commented:
Added users
- Login only allows authenticated users to login 
- Redirects user to main page after login
- Local storage used for user state to persist login between sessions/loads
- Small page displaying currently logged in user info and logout button
- Comments can only be deleted by users who posted them 
- Comments can only be posted by logged in users 
- Gives link to user to login if they want to add comment without being logged in
- Added header section so user knows they are logged in/who they are logged in as

View/Add comments
- Added toggle buttons for both viewing comments and adding a comment to a post
- Added section above comments so user knows what their username is before commenting

Errors (task 12) - I got carried away
- Added error endpoints for 
- Unknow endpoints 
- Reviews that don't exist
- Categories that don't exist

dappernerddesigns commented:
I do like the messages to the user to show a comment has been posted and deleted, these do persist on screen though. While not an issue, it would be nice to see these disappear after a few seconds. 

Nice error pages, but do try to keep one PR for one functionality. Naughty hobbit
 
Merge when ready
